exercicios/para-casa/resolucao_exercicio.py

```diff
@@ -0,0 +1,146 @@
+    {'ano_receita': 2024, 'mes_receita': '12', 'faturamento': 39303, 'despesas': 14418}
+]
+
+
+def menu():
+    while True:
+        print("-------- MENU --------")
+        print("Digite 1 para adicionar um novo registro")
+        print("Digite 2 para consultar a receita de certo ano")
+        print("Digite 3 para consultar quais os anos com a pior e melhor receita")
+        print("Digite 0 para sair")
+
+        opcao = input("Insira a opção escolhida: ")
+
+        if opcao != "1" and opcao != "2" and opcao != "3" and opcao != "0":
+            print("Opção inválida!")
+            continue
+
+        if opcao == "1":
+            adicionar_registro()
+        elif opcao == "2":
+            consultar_receita()
+        elif opcao == "3":
+            melhor_pior_ano()
+        elif opcao == "0":
+            break
+
+
+def adicionar_registro():
+    ano = int(input("Insira o ano do registro: "))
+    mes = input("Insira o mês do registro: ")
+    lucro = int(input(f"Insira o faturamento do mês {mes} de {ano}: "))
+    gasto = int(input(f"Insira as despesas do mês {mes} de {ano}: "))
+
+    novo_registro = {
+        "ano_receita": ano,
+        "mes_receita": mes,
+        "faturamento": lucro,
+        "despesas": gasto
+    }
+
+    dataset.append(novo_registro)
+
+
+def consultar_receita():
+    ano = int(input("Insira o ano do qual deseja consultar a receita: "))
+    receita_total = 0
+    maior_lucro = 0
+    maior_despesa = 0
+    mes_lucrativo = ''
+    mes_prejuizo = ''
+
+    for registro in dataset:
+        if ano == registro.get("ano_receita"):
+            receita = registro.get("faturamento") - registro.get("despesas")
+            receita_total += receita
+        
+            if registro.get("faturamento") >= maior_lucro:
+                maior_lucro = registro.get("faturamento")
+                mes_lucrativo = registro.get("mes_receita")
+        
+            if registro.get("despesas") >= maior_despesa:
+                maior_despesa = registro.get("despesas")
+                mes_prejuizo = registro.get("mes_receita") 
+    
+    print(f"A receita total em {ano} foi de R${receita_total:.2f}.")
+    print(f"O mês mais lucrativo do ano de {ano} foi o mês {mes_lucrativo}, com um faturamento de {maior_lucro}")
+    print(f"O mês com mais despesas do ano de {ano} foi o mês {mes_prejuizo}, com gastos equivalentes a {maior_despesa}")
+
+
+def melhor_pior_ano():
+    melhor_receita = 0
+    pior_receita = 0
+    anos_registrados = []
+    melhor_ano = 0
+    pior_ano = 0
+
+    for registro in dataset:
+        if registro.get("ano_receita") not in anos_registrados:
+            anos_registrados.append(registro.get("ano_receita"))
+
+    receitas = []
+
+    for ano in anos_registrados:
+        receita_anual = 0
+
+        for registro in dataset:
+            if registro.get("ano_receita") == ano:
+                receita_anual += (registro.get("faturamento") - registro.get("despesas"))
```

Comment:
muito bem!

Uma curiosidade: colocar `for` dentro de `for` não é muito indicado já que normalmente tem uma performance ruim.